### PR TITLE
feat(BBD-833): Connect CVS Past Purchase work to live API

### PR DESCRIPTION
**Pull request by zfcb**
groupby/flux-capacitor#139
groupby/storefront-core#150
groupby/storefront-products#28
groupby/storefront-sort#18
groupby/storefront-structure#41
groupby/storefront-navigation#32
groupby/storefront-recommendations#8
https://github.com/groupby/storefront-presets/pull/52

### CODE REVIEW

File: src/sort/index.ts [hunk 1]
```diff
@@ -1,33 +1,57 @@
...
 
 @configurable
 @alias('sort')
 @tag('gb-sort', require('./index.html'))
 class Sort {
 
   props: Sort.Props = {
-    labels: []
+    labels: [],
   };
```
Comment (treeforever) [inline src/sort/index.ts:10]: A point for discussion: maybe we can set a default value for props.storeSection?

File: test/unit/sort.ts [hunk 2]
```diff
@@ -91,6 +115,28 @@ suite('Sort', ({ expect, spy, stub, itShouldBeConfigurable, itShouldHaveAlias })
...
+      const getPastPurchasesLabel = sort.getPastPurchasesLabel = spy(() => 'x');
+      const sort1 = { field: 'variant.colour' };
+      const sort2 = { field: 'price' };
+      const sort3 = { field: 'size' };
+      const select = sort.select = spy(() => ({ items: [sort1, sort2, sort3], selected: 1 }));
+      sort.props.storeSection = StoreSections.PAST_PURCHASES;
+
+      const options = sort.extractSorts();
+
+      expect(select).to.be.calledWithExactly(Selectors.pastPurchaseSort);
```
Comment (treeforever) [inline test/unit/sort.ts:130]: stub Selectors.pastPurchaseSort and use its value for calledWithExactly.

File: src/sort/index.ts (hunk 1 above)
Comment (Filip-F) [reply src/sort/index.ts:10]: We do, this happens in core, and is inherited by the components.

File: test/unit/sort.ts (hunk 2 above)
Comment (Filip-F) [reply test/unit/sort.ts:130]: We use this.select so it should be tested this way.

File: src/sort/index.ts [hunk 3]
```diff
@@ -1,33 +1,57 @@
...
+        case StoreSections.SEARCH:
+          this.actions.selectSort(index);
+      }
+    }
   };
 
   init() {
     this.updateSorts();
-    this.flux.on(Events.SORTS_UPDATED, this.updateSorts);
+    switch (this.props.storeSection) {
```
Comment (effervescentia) [inline src/sort/index.ts:26]: can we abstract this switch logic in the short term if we have to keep reusing it in all the components?

**Review by effervescentia — CHANGES_REQUESTED** [review]

File: src/sort/index.ts (hunk 3 above)
Comment (Filip-F) [reply src/sort/index.ts:26]: I don't think the switch logic could easily be abstracted out without refactoring the selectors/events entirely as only the component can know what store sections relate to what actions need to be listened to.

**Review by effervescentia — APPROVED** [review]

**Pull request merged**